### DISCUSSION TITLE
Using the destroySession function, when token validation fails

### DISCUSSION THREAD

**Discussion by MatD1** [sergiodxa/remix-auth#194] [Q&A] [answered]
I am trying to use the destroySession function, to destroy the user's session as this is a requirement of using the Twitch API and preferably I'd like to log them in again, however running into some trouble, any help is greatly appreciated.

#### I am attempting to use React-Query to re-fetch the endpoint every hour, however, I am open to better suggestions.

```
  const { error, data } = useQuery("TV-Revalidate", () =>
    fetch("https://id.twitch.tv/oauth2/validate", {
      headers: {
        Authorization: `Bearer ${user?.token}`,
      },
    }).then((res) => res.json())
  );

  // if (error) return destroySession;

  if (data?.status === 401)
    return async () => {
      await destroySession()
    };
```
This fails to completely render because React doesn't approve of it being used as a child inside of a function, how I am supposed to pass the session argument?
<img width="594" alt="Screen Shot 2022-09-21 at 12 12 56 am" src="https://user-images.githubusercontent.com/64875762/191281304-5b66f115-9984-4daa-9859-fd6124553b83.png">

**Comment by sergiodxa** [comment, marked as answer]
Since you're using useQuery I suppose you're trying to do this client-side, and destroySession can't be used client-side, it's a server-side only thing.

What you could do is to send a POST to some resource route and call destroy session there.

BTW, this is not related to Remix Auth.
